### PR TITLE
Functionality added for GCS delete multipart upload policy

### PR DESCRIPTION
**Pull request by neerajsinghal05**
Functionality added for GCS delete multipart upload policy

### CODE REVIEW

File: src/main/java/io/cdap/e2e/utils/StorageClient.java [hunk 1]
```diff
@@ -40,7 +45,8 @@ public class StorageClient {
 
   private static Storage getStorageService() throws IOException {
     return (null == storageService) ? StorageOptions.newBuilder().setProjectId(
-      PluginPropertyUtils.pluginProp(ConstantsUtil.PROJECT_ID)).build().getService() : storageService;
+        PluginPropertyUtils.pluginProp(ConstantsUtil.PROJECT_ID)).build().getService()
+        : storageService;
```
Comment (rahuldash171) [inline src/main/java/io/cdap/e2e/utils/StorageClient.java]: why the writing format has been changed if there is no change ?

File: src/main/java/io/cdap/e2e/utils/StorageClient.java [hunk 2]
```diff
@@ -52,7 +58,7 @@ public static Page<Blob> listObjects(String bucketName) throws IOException {
   }
 
   public static Page<Blob> listObjectsWithPrefix(
-    String bucketName, String directoryPrefix) throws IOException {
+      String bucketName, String directoryPrefix) throws IOException {
```
Comment (rahuldash171) [inline src/main/java/io/cdap/e2e/utils/StorageClient.java]: remove space so that it can not be seen as a new line

File: src/main/java/io/cdap/e2e/utils/StorageClient.java [hunk 3]
```diff
@@ -66,12 +72,14 @@ public static boolean deleteObject(String bucketName, String objectName) throws
   }
 
   public static Blob getObjectMetadata(String bucketName, String blobName)
-    throws StorageException, IOException {
-    return getStorageService().get(bucketName, blobName, Storage.BlobGetOption.fields(Storage.BlobField.values()));
+      throws StorageException, IOException {
+    return getStorageService().get(bucketName, blobName,
+        Storage.BlobGetOption.fields(Storage.BlobField.values()));
   }
```
Comment (rahuldash171) [inline src/main/java/io/cdap/e2e/utils/StorageClient.java]: remove new line , no change here too, work on spaces . format should not be changed for existing code

File: src/main/java/io/cdap/e2e/utils/StorageClient.java [hunk 4]
```diff
@@ -66,12 +72,14 @@ public static boolean deleteObject(String bucketName, String objectName) throws
...
 
   public static Bucket getBucketMetadata(String bucketName) throws IOException {
-    return getStorageService().get(bucketName, Storage.BucketGetOption.fields(Storage.BucketField.values()));
+    return getStorageService().get(bucketName,
+        Storage.BucketGetOption.fields(Storage.BucketField.values()));
```
Comment (rahuldash171) [inline src/main/java/io/cdap/e2e/utils/StorageClient.java]: Same here , dont change existing code

File: src/main/java/io/cdap/e2e/utils/StorageClient.java [hunk 5]
```diff
@@ -87,10 +95,18 @@ public static Bucket createBucket(String bucketName) throws IOException {
   }
 
   public static Blob uploadObject(String bucketName, String objectName, String filePath)
-    throws IOException, URISyntaxException {
+      throws IOException, URISyntaxException {
```
Comment (rahuldash171) [inline src/main/java/io/cdap/e2e/utils/StorageClient.java]: Same here , undo the changes

File: src/main/java/io/cdap/e2e/utils/StorageClient.java [hunk 6]
```diff
@@ -40,7 +45,8 @@ public class StorageClient {
 
   private static Storage getStorageService() throws IOException {
     return (null == storageService) ? StorageOptions.newBuilder().setProjectId(
-      PluginPropertyUtils.pluginProp(ConstantsUtil.PROJECT_ID)).build().getService() : storageService;
+        PluginPropertyUtils.pluginProp(ConstantsUtil.PROJECT_ID)).build().getService()
```
Comment (itsmekumari) [inline src/main/java/io/cdap/e2e/utils/StorageClient.java]: If there is no change in code, revert them this should not be shown in PR.

File: src/main/java/io/cdap/e2e/utils/StorageClient.java [hunk 7]
```diff
@@ -87,10 +95,18 @@ public static Bucket createBucket(String bucketName) throws IOException {
   }
 
   public static Blob uploadObject(String bucketName, String objectName, String filePath)
-    throws IOException, URISyntaxException {
+      throws IOException, URISyntaxException {
     return getStorageService().create(
-      BlobInfo.newBuilder(BlobId.of(bucketName, objectName)).build(),
-      Files.readAllBytes(Paths.get(StorageClient.class.getResource("/" + filePath).toURI())));
+        BlobInfo.newBuilder(BlobId.of(bucketName, objectName)).build(),
+        Files.readAllBytes(Paths.get(StorageClient.class.getResource("/" + filePath).toURI())));
```
Comment (rahuldash171) [inline src/main/java/io/cdap/e2e/utils/StorageClient.java]: Same here , undo the changes

File: src/main/java/io/cdap/e2e/utils/StorageClient.java [hunk 8]
```diff
@@ -87,10 +95,18 @@ public static Bucket createBucket(String bucketName) throws IOException {
...
   }
 
+  public static Bucket createBucketwithLifeCycle(String bucketName, int age) throws IOException {
+    return getStorageService().create(BucketInfo.of(bucketName)).toBuilder()
+        .setLifecycleRules(
+            ImmutableList.of(
+                new LifecycleRule(
+                    LifecycleAction.newAbortIncompleteMPUploadAction(),
+                    LifecycleCondition.newBuilder().setAge(age).build()))).build().update(); }
```
Comment (rahuldash171) [inline src/main/java/io/cdap/e2e/utils/StorageClient.java]: Add the curly braces below and there should be a space between the final and  this curly braces after you change .
     LifecycleCondition.newBuilder().setAge(age).build()))).build().update();
    }

}

**Review by rahuldash171 — COMMENTED** [review]
Please address review comments

File: src/main/java/io/cdap/e2e/utils/StorageClient.java (hunk 3 above)
Comment (neerajsinghal05) [reply src/main/java/io/cdap/e2e/utils/StorageClient.java]: Done

File: src/main/java/io/cdap/e2e/utils/StorageClient.java (hunk 7 above)
Comment (neerajsinghal05) [reply src/main/java/io/cdap/e2e/utils/StorageClient.java]: Done

File: src/main/java/io/cdap/e2e/utils/StorageClient.java (hunk 8 above)
Comment (rahuldash171) [reply src/main/java/io/cdap/e2e/utils/StorageClient.java]: resolve this .

File: src/main/java/io/cdap/e2e/utils/StorageClient.java (hunk 7 above)
Comment (rahuldash171) [reply src/main/java/io/cdap/e2e/utils/StorageClient.java]: change is not seen it is still seen as new line

File: src/main/java/io/cdap/e2e/utils/StorageClient.java [hunk 9]
```diff
@@ -40,7 +45,7 @@ public class StorageClient {
 
   private static Storage getStorageService() throws IOException {
     return (null == storageService) ? StorageOptions.newBuilder().setProjectId(
-      PluginPropertyUtils.pluginProp(ConstantsUtil.PROJECT_ID)).build().getService() : storageService;
+        PluginPropertyUtils.pluginProp(ConstantsUtil.PROJECT_ID)).build().getService() : storageService;
```
Comment (rahuldash171) [inline src/main/java/io/cdap/e2e/utils/StorageClient.java]: new line is seeen here too

File: src/main/java/io/cdap/e2e/utils/StorageClient.java (hunk 6 above)
Comment (neerajsinghal05) [reply src/main/java/io/cdap/e2e/utils/StorageClient.java]: Changes has been done , as per the review comment

File: src/main/java/io/cdap/e2e/utils/StorageClient.java (hunk 1 above)
Comment (neerajsinghal05) [reply src/main/java/io/cdap/e2e/utils/StorageClient.java]: Changes has been done , as per the review comment

File: src/main/java/io/cdap/e2e/utils/StorageClient.java (hunk 2 above)
Comment (neerajsinghal05) [reply src/main/java/io/cdap/e2e/utils/StorageClient.java]: Changes has been done , as per the review comment

File: src/main/java/io/cdap/e2e/utils/StorageClient.java (hunk 4 above)
Comment (neerajsinghal05) [reply src/main/java/io/cdap/e2e/utils/StorageClient.java]: Changes has been done , as per the review comment

File: src/main/java/io/cdap/e2e/utils/StorageClient.java (hunk 7 above)
Comment (neerajsinghal05) [reply src/main/java/io/cdap/e2e/utils/StorageClient.java]: Changes has been done , as per the review comment

File: src/main/java/io/cdap/e2e/utils/StorageClient.java (hunk 8 above)
Comment (neerajsinghal05) [reply src/main/java/io/cdap/e2e/utils/StorageClient.java]: Changes has been done , as per the review comment

File: src/main/java/io/cdap/e2e/utils/StorageClient.java (hunk 9 above)
Comment (neerajsinghal05) [reply src/main/java/io/cdap/e2e/utils/StorageClient.java]: Changes has been done , as per the review comment

**Review by rahuldash171 — APPROVED** [review]
Approved .
